Add analyzer error message when Policies using JWT are not configured properly

jacob-delgado opened the pull request:
When a v1alpha1 Authentication Policy using JWT has a misconfigured K8s Service we wish for the analyzer to inform the user that they have misconfigured JWT authentication. The work to implement the analyzer using this message will be done in a separate PR.

This is to address issue https://github.com/istio/istio/issues/17535.

For the discussion about this analyzer see
https://github.com/istio/istio/pull/20450
https://github.com/istio/istio/pull/20645

istio-testing commented:
Hi @jacob-delgado. Thanks for your PR.

I'm waiting for a [istio](https://github.com/orgs/istio/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/istio/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands?repo=istio%2Fistio).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

esnible commented:
/ok-to-test

istio-testing commented:
@jacob-delgado: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-mixer-k8s-tests_istio | 7d9ad0e415393335629ff6c7f7860eb3da68e756 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/20671/integ-mixer-k8s-tests_istio/5782) | `/test integ-mixer-k8s-tests_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@jacob-delgado: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-mixer-k8s-tests_istio | 7d9ad0e415393335629ff6c7f7860eb3da68e756 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/20671/integ-mixer-k8s-tests_istio/5782) | `/test integ-mixer-k8s-tests_istio`
integ-istioio-k8s-tests_istio | 7d9ad0e415393335629ff6c7f7860eb3da68e756 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/20671/integ-istioio-k8s-tests_istio/4155) | `/test integ-istioio-k8s-tests_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

jacob-delgado commented:
@esnible I believe I addressed your comments by adding a comment as to why it is being used without any code.

The analyzer issue is to fix https://github.com/istio/istio/issues/17535 and the deprecation message for istio 1.5 and beyond to inform users about new thew api is found at https://github.com/istio/istio/pull/20645.

Both of these issues will need to be cherry-picked into the release-1.5 branch. Getting this merged into master and cherry-picking into release-1.5 will allow me to unblock the work as defined https://github.com/istio/istio/pull/20672#issuecomment-580508680 to get working

istio-testing commented:
@jacob-delgado: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
gencheck_istio | cf3a2c1f87df1a5f169dd1d3f230543679d3742c | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/20671/gencheck_istio/6104) | `/test gencheck_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

jacob-delgado commented:
/retest

nrjpoddar commented:
/cherrypick/release-1.5
